lib/dry/auto_inject/builder.rb

```diff
@@ -15,10 +15,17 @@ class Builder < BasicObject
       def initialize(container, options = {})
         @container = container
         @strategies = options.fetch(:strategies) { Strategies }
+        @pattern = options.fetch(:pattern) { nil }
+      end
+
+      def allow(pattern)
+        AutoInject::Builder.new(@container, pattern: pattern, strategies: @strategies)
       end
 
       # @api public
       def [](*dependency_names)
+        raise unless match_by_pattern?(dependency_names)
```

Comment:
I think, it will be better to raise some fixed error, not `RuntimeError`. I think, it would be great for programmers. WDYT about it?